hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/fsdataset/impl/ReplicaMap.java

```diff
@@ -238,9 +221,10 @@ ReplicaInfo remove(String bpid, long blockId) {
    * @return the number of replicas in the map
    */
   int size(String bpid) {
+    LightWeightResizableGSet<Block, ReplicaInfo> m = null;
     try (AutoCloseableLock l = readLock.acquire()) {
-      FoldedTreeSet<ReplicaInfo> set = map.get(bpid);
-      return set != null ? set.size() : 0;
+      m = map.get(bpid);
```

Comment:
The definition of `m` can be moved into the try clause.
```suggestion
      GSet<Block, ReplicaInfo> m = map.get(bpid);
```